### PR TITLE
[IAST - Insecure hashing] New endpoints spring-boot(java) and express…

### PR DESCRIPTION
**Pull request by robertomonteromiguel**
Add new endpoints to check insecure hash IAST
Add new test to check that endpoints are available (Does not check for vulnerability detection as long as it is not implemented in the library.)

### CODE REVIEW

File: utils/build/docker/java/spring-boot/src/main/java/com/datadoghq/system_tests/springboot/iast/utils/CryptoExamples.java [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+
+		public static Stream<InsecureHashingAlgorithm> stream() {
+			return Stream.of(InsecureHashingAlgorithm.values());
+		}
+
+		public String getAlgorithmName() {
+			return this.algorithm;
+		}
+	}
+}
```
Comment (vdeturckheim) [inline utils/build/docker/java/spring-boot/src/main/java/com/datadoghq/system_tests/springboot/iast/utils/CryptoExamples.java:61]:
```suggestion
}

```

File: utils/build/docker/nodejs/express4/app.js [hunk 2]
```diff
@@ -52,6 +52,18 @@ app.get('/identify', (req, res) => {
...
+app.get('/iast/insecure_hashing', (req, res) => {
+  const span = tracer.scope().active();
+  span.setTag('appsec.event"', true);
+  var crypto = require('crypto');
+  var name = 'insecure';
+  var hashMd5 = crypto.createHash('md5').update(name).digest('hex');
+  var hashMd4 = crypto.createHash('md4').update(name).digest('hex');
+  var hashSha1 = crypto.createHash('sha1').update(name).digest('hex');
+  var outputHashes = "MD5:".concat(hashMd5).concat("----").concat("MD4:").concat(hashMd4).concat("----").concat("sha1:").concat(hashSha1);
+  console.log(outputHashes);
```
Comment (vdeturckheim) [inline utils/build/docker/nodejs/express4/app.js:64]:
```suggestion
```
Do we need a log here?

File: utils/build/docker/nodejs/express4/app.js [hunk 3]
```diff
@@ -52,6 +52,18 @@ app.get('/identify', (req, res) => {
 app.get('/status', (req, res) => {
   res.status(parseInt(req.query.code)).send('OK');
 });
+app.get('/iast/insecure_hashing', (req, res) => {
+  const span = tracer.scope().active();
+  span.setTag('appsec.event"', true);
+  var crypto = require('crypto');
```
Comment (uurien) [inline utils/build/docker/nodejs/express4/app.js:58]: You can use `const` instead of `var` for all `var` in those lines.

**Review by cbeauchesne — CHANGES_REQUESTED** [review]
Could you update `docs/weblog/README.md`, and details what this endpoint is supposed to do ?

File: utils/build/docker/java/spring-boot/src/main/java/com/datadoghq/system_tests/springboot/AppSecIast.java [hunk 4]
```diff
@@ -0,0 +1,30 @@
...
+	String insecureHashing() {
+		
+		String superSecretAccessKey = "insecure";
+		
+		final Span span = GlobalTracer.get().activeSpan();
+		if (span != null) {
+			span.setTag("appsec.event", true);
+		}
+
+		return CryptoExamples.InsecureHashingAlgorithm.stream()
```
Comment (smola) [inline utils/build/docker/java/spring-boot/src/main/java/com/datadoghq/system_tests/springboot/AppSecIast.java:25]: I wonder if we want to pass the algorithm name (e.g. `md5`) in the request to test them independently?

**Review by smola — APPROVED** [review]
Sounds good. I just added a suggestion to test algorithms in independent requests. This would be particularly useful if/when we limit the number of reported vulnerabilities per request. Also to verify that every algorithm is checked (once we test for the vulnerability detection).

**Review by cbeauchesne — APPROVED** [review]

**Review by cbeauchesne — APPROVED** [review]

**Comment by cbeauchesne** [comment]
@robertomonteromiguel you can merge it when you want

**Pull request merged**